crates/key-value/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+use anyhow::Result;
+use std::collections::HashMap;
+use table::Table;
+use wit_bindgen_wasmtime::async_trait;
+
+mod host_component;
+pub mod sqlite;
+mod table;
+
+pub use host_component::KeyValueComponent;
+
+wit_bindgen_wasmtime::export!({paths: ["../../wit/ephemeral/key-value.wit"], async: *});
+
+use key_value::{Error, KeyValue, Store};
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+impl std::error::Error for Error {}
+
+#[derive(Clone)]
+pub enum ImplConfig {
+    Sqlite(sqlite::DatabaseLocation),
+}
+
+#[derive(Clone)]
+pub struct Config {
+    pub configs: HashMap<String, ImplConfig>,
+}
+
+#[async_trait]
+trait Impl: Sync + Send {
```

Comment:
The trait names here, `Impl` and `ImplStore` are fairly unintuitive. Is this a naming pattern inherited from other libraries?